### DISCUSSION TITLE
If I call h5gopen, and it fails, do i still need to call h5gclose after to release memory??

### DISCUSSION THREAD

**Discussion by jjepson2** [HDFGroup/hdf5#3627] [Q&A] [answered]
If the return error is that of a failure from h5gopen, do I still need to call h5gclose afterwards to release potential memory allocations?? And the same question for h5[*]open (such as for opening properties, attributes, etc.).

**Comment by jhendersonHDF** [comment, marked as answer]
Hi @jjepson2,

For any object (file, group, dataset, attribute, etc.) that you attempt to open,
if the H5Xopen (H5Fopen, H5Dopen, etc.) call fails, then you do NOT need to
call H5Xclose to release any potential memory allocations; the library should
do this and return H5I_INVALID_HID from the H5Xopen call. Attempting to call
H5Xclose on that returned ID value will cause an error from HDF5.

However, if you find that memory is still allocated after an H5Xopen call fails,
please let us know so that we can investigate.

**Reply by jjepson2** [reply]
Thank you.  That is what I was wanting to know.
